python/pyspark/pandas/frame.py

```diff
@@ -3989,9 +3989,14 @@ def insert(
             )
 
         if is_name_like_tuple(column):
-            if len(column) != len(self.columns.levels):
-                # To be consistent with pandas
-                raise ValueError('"column" must have length equal to number of column levels.')
+            if isinstance(self.columns, pd.MultiIndex):
```

Comment:
> pandas API on Spark doesn't support assigning the tuple-like column name to the DataFrame as below:

Seems like this is matched with the pandas behaviour. can we make it matched?

```python
>>> import pandas as pd
>>> pdf = pd.DataFrame({'a': [1,2,3]})
>>> pdf
   a
0  1
1  2
2  3
>>> pdf.insert(0, ("a", "b"), [1,2,3])
>>> pdf
   (a, b)  a
0       1  1
1       2  2
2       3  3
```

Reply:
Pandas supports any `Hashable` object here.

Reply:
Yeah, assigning the tuple as column name is only supported for MultiIndex columns for now